src/api/c/graphics_common.cpp

```diff
@@ -178,15 +233,15 @@ void ForgeManager::setWindowChartGrid(const forge::Window* window,
 {
     ChartMapIter iter = mChartMap.find(window);
 
-        if(iter != mChartMap.end()) {
-
-    }
-
     if(iter != mChartMap.end()) {
         // ChartVec found. Clear it.
         // TODO: Should we clear this even if r = old_r and c = old_c?
-        for(int i = 0; i < (int)(iter->second).size(); i++)
-            if((iter->second)[i] != NULL) delete (iter->second)[i];
+        for(int i = 0; i < (int)(iter->second).size(); i++) {
```

Comment:
We can avoid clearing if grid size is same but i believe we should reset the chart properties to default because the new charts might be of same properties.